src/installer/pkg/sfx/installers/dotnet-host.proj

```diff
@@ -78,4 +78,35 @@
       <RpmJsonProperty Include="@(PackageConflictsProperty)" />
     </ItemGroup>
   </Target>
+
+  <PropertyGroup>
+    <_MacOSIntermediatesPath>$(IntermediateOutputPath)macos/</_MacOSIntermediatesPath>
+    <!-- CreatePkg depends on this property being set.-->
+    <MacOSScriptsDirectory>$(_MacOSIntermediatesPath)scripts</MacOSScriptsDirectory>
+  </PropertyGroup>
+  <ItemGroup>
+    <_MacOSScript Include="$(MacOSScriptsTemplateDirectory)/*" Destination="$(MacOSScriptsDirectory)/%(FileName)%(Extension)"/>
+  </ItemGroup>
+  <Target Name="ReplaceTemplateParametersInMacOsScripts" 
+          Inputs="@(_MacOSScript)"
+          Outputs="@(_MacOSScript->'%(Destination)')"
+          DependsOnTargets="_GetInstallerProperties"
+          BeforeTargets="CreatePkg">
+
+    <PropertyGroup>
+      <_UnameMachineRegex>$(InstallerTargetArchitecture)</_UnameMachineRegex>
+      <_UnameMachineRegex Condition="'$(InstallerTargetArchitecture)' == 'x64'">x64|x86_x64</_UnameMachineRegex>
```

Comment:
```suggestion
      <_UnameMachineRegex Condition="'$(InstallerTargetArchitecture)' == 'x64'">x64|x86_64</_UnameMachineRegex>
```

Reply:
Thanks, I was looking for official docs on this but had a hard time finding them.  I only had a M1 to test on.  Do you happen to know where the docs are?  I'm wondering if we need to handle more values.

Reply:
Like aarch64 for arm, for example.  I only see the M1 return `arm64` but I'd really like to see docs on that to be certain.

Reply:
I am using BigSur x64, tested with `uname -m` which returns: `x86_64`. I think there is no agreement on these values across various Unix like OS. The possible values are `x86_64` and `amd64`. Some related resources:

* https://github.com/dotnet/runtime/blob/3409f73c7893f863297e5eb5d17241dcf0796733/eng/native/init-os-and-arch.sh
* CMake uses heuristics based on different `uname` flags, e.g. https://github.com/Kitware/CMake/blob/20b2bf00479ff656d0f6b772ba5673d1cb7007c4/Modules/CMakeDetermineSystem.cmake#L60
* https://stackoverflow.com/a/45125525/863980

Reply:
I see, we check a these values:
https://github.com/dotnet/runtime/blob/3409f73c7893f863297e5eb5d17241dcf0796733/eng/native/init-os-and-arch.sh#L36-L40

I might match those for consistency in the repo.

I see wikipedia catalogs the values:
https://en.wikipedia.org/wiki/Uname
Looks like macos never changed the value once they started, so that's promising (this script only needs to run on macos).

None of these resources are what I would consider primary sources but it's better than nothing.  Thanks for sharing!